### PR TITLE
Adding OpenStack cloud support

### PR DESCRIPTION
**Pull request by mstair**
Addition of full OpenStack cloud support. Compliments the related cloudaux [PR](https://github.com/Netflix-Skunkworks/cloudaux/pull/48). Initial commit includes watchers for many of the neutron networking services (security groups, networks, ports, routers, subnets), nova instances and swift object containers (OpenStack version of S3 buckets). Initial auditors are for security group (which inherits from AWS secgroup) and object container ACLs. 

The code is a hybrid of AWS and GCP security monkey conventions. Given users can be at various releases of OpenStack, even potentially various version of the different services, the code takes advantage of the openstack client generators to dynamically build the item configs. I have tested against a few older OpenStack versions (Juno/Mitaka) and recent (Pike). I plan to test against Rackspace (do not currently have access to an account).

Documentation was updated and unit tests were created for all watcher/auditors and pass in travis against my fork.

### CODE REVIEW

**Comment by mstair** [comment]
Did a pretty major merge with develop since I submitted this PR, let me know if any issues.

**Comment by monkeysecurity** [comment]
I've scheduled a code-review with my team on this for 11/1.  Hopefully we'll have this merged pretty soon.

**Comment by mstair** [comment]
Not sure if you had time to review the PR code, please let me know if questions or concerns.

File: security_monkey/watchers/openstack/openstack_watcher.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        self.honor_ephemerals = True
+        self.ephemeral_paths = [ "updated_at" ]
+
+    def _get_openstack_creds(self, account):
+        from security_monkey.datastore import Account
+
+        _account = Account.query.filter(Account.name.in_([account])).one()
+        return _account.identifier, _account.getCustom('cloudsyaml_file')
+
+    """ Regions are not global but account specific """
```
Comment (monkeysecurity) [inline security_monkey/watchers/openstack/openstack_watcher.py]: I thought docstrings went after the method signature.  Does this work?  Is it still called a docstring?

File: security_monkey/watchers/openstack/openstack_watcher.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                return results, exception_map
+
+            for item in item_list:
+                item_name = self.get_name_from_list_output(item)
+                if item_name and self.check_ignore_list(item_name):
+                    continue
+
+                item_details = invoke_get_method(item, **kwargs)
+                if item_details:
+                    arn = 'arn:openstack:{region}:{cloud_name}:{item_type}/{item_id}'.format(
```
Comment (monkeysecurity) [inline security_monkey/watchers/openstack/openstack_watcher.py]: nice

File: README.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ Security Monkey
 
 <img align="right" alt="Security Monkey Logo 2017" src="docs/images/Security_Monkey.png" width="50%">
 
-Security Monkey monitors your [AWS and GCP accounts](https://medium.com/@Netflix_Techblog/netflix-security-monkey-on-google-cloud-platform-gcp-f221604c0cc7) for policy changes and alerts on insecure configurations.  It provides a single UI to browse and search through all of your accounts, regions, and cloud services.  The monkey remembers previous states and can show you exactly what changed, and when.
+Security Monkey monitors your [AWS and GCP accounts](https://medium.com/@Netflix_Techblog/netflix-security-monkey-on-google-cloud-platform-gcp-f221604c0cc7) for policy changes and alerts on insecure configurations.  OpenStack support is available for Rackspace and private clouds.  It provides a single UI to browse and search through all of your accounts, regions, and cloud services.  The monkey remembers previous states and can show you exactly what changed, and when.
```
[the remark this answers is not in the document]
Comment (mstair) [reply README.md]: No, you're right, it should be generic (Rackspace on brain from one of our projects).

File: security_monkey/watchers/openstack/openstack_watcher.py (hunk 1 above)
Comment (mstair) [reply security_monkey/watchers/openstack/openstack_watcher.py]: No, got comment happy, I will fix.

**Pull request merged**